.github/workflows/plan-terraform.yaml

```diff
@@ -53,6 +57,8 @@ jobs:
           juju show-controller $(echo $CONTROLLER|tr -d '"') | yq '.[$CONTROLLER]'.details.\"ca-cert\"|tr -d '"'
           echo EOF
         } >> "$GITHUB_ENV"
+
+        sed -i "s#ref=[^&\"]*#ref=${{ github.head_ref }}#" ${{env.WORKING_DIR}}/test.tf
```

Comment:
we switch the `test.tf` to confirm it's testing from this source branch during a pull request.